File: 프로그래머스/1/131705. 삼총사/삼총사.swift
import Foundation

func solution(_ number:[Int]) -> Int {
    var result = 0
    let size = number.count
    
    for i in 0..<(size-2) {
        for j in i+1..<(size-1) {
            for k in j+1..<(size) {
                if (number[i] + number[j] + number[k]) == 0 { result += 1}
            }
        }
    }
    return result
}

File: 프로그래머스/1/131705. 삼총사/삼총사.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> number) {
    int answer = 0;
    int size = number.size();

    for (int i = 0; i < size; i++) {
        for (int j = i + 1; j < size; j++) {
            for (int k = j + 1; k < size; k++) {
                if (number[i] + number[j] + number[k] == 0) {
                    answer++;
                }
            }
        }
    }

    return answer;
}